lib/faraday/parameters.rb

```diff
@@ -39,13 +39,15 @@ def self.encode(params)
             buffer << "#{to_query.call(new_parent, val)}&"
           end
           return buffer.chop
-        elsif value.is_a?(Array)
+        elsif value.is_a?(Array) && value.size != 0
           buffer = ""
           value.each_with_index do |val, i|
             new_parent = "#{parent}%5B%5D"
             buffer << "#{to_query.call(new_parent, val)}&"
           end
           return buffer.chop
+        elsif value.is_a?(Array) && value.size == 0
```

Comment:
Perhaps the `Array#empty?` method is even clearer to read?

Reply:
I'd go a step further and merge this 2 branches:

```ruby
elsif value.is_a?(Array)
  return "#{parent}%5B%5D" if value.empty? # This is the only new line
  buffer = ""
  value.each_with_index do |val, i|
    new_parent = "#{parent}%5B%5D"
     buffer << "#{to_query.call(new_parent, val)}&"
  end
  return buffer.chop
```

Reply:
nice catch
fixed